DataFormats/TrackReco/src/HitPattern.cc

```diff
@@ -142,6 +143,13 @@ uint16_t HitPattern::encode(const DetId &id, TrackingRecHit::Type hitType)
                 layer |= abs(rpcid.region());
             }
             break;
+         case MuonSubdetId::GEM:
+            {
+              GEMDetId gemid(id.rawId());
+              layer = ((gemid.roll()-1)<<1);
```

Comment:
packing should probably be reconsidered.
The current implementation doesn't work for GE2/1